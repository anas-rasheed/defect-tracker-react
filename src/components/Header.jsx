import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledNavbar = styled.nav`
  height: 60px;
  display: flex;
  align-items: center;
  background: linear-gradient(
    to right bottom,
    rgb(255, 255, 255, 0.7),
    rgb(255, 255, 255, 0.3)
  );
  backdrop-filter: blur(100rem);
  box-shadow: 0px 6px 20px rgb(157, 160, 252, 1);
`;

const Header = () => {
  return (
    <>
      <StyledNavbar>
        <Link to='/postingRouter'>Brand</Link>
      </StyledNavbar>
    </>
  );
};

export default Header;
