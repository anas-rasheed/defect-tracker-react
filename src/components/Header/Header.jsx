import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { logout } from '../../redux/slices/authSlice';

const StyledNavbar = styled.nav`
  height: 60px;
  display: flex;
  align-items: center;
  background: linear-gradient(
    to right bottom,
    rgba(255, 255, 255, 0.7),
    rgba(255, 255, 255, 0.3)
  );
  box-shadow: 0px 6px 20px rgba(157, 160, 252, 1);
  border-radius: 2rem;
  position: relative;
  justify-content: space-evenly;
`;

const StyledBrand = styled.span`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  position: absolute;
  left: 20px;
  > a {
    color: #ffffff;
    text-decoration: none;
    padding: 10px;
    background-color: rgba(0, 0, 0, 0.7);
    border-radius: 2rem;
  }
`;

const StyledNavLinks = styled.span`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  position: absolute;
  right: 10px;
  > a {
    color: #ffffff;
    text-align: center;
    margin: 0 5px;
    // width: 50px;
    text-decoration: none;
    padding: 10px;
    background-color: rgba(0, 0, 0, 0.7);
    border-radius: 2rem;
  }
`;

export const Header = ({ isLoggedIn, logout }) => {
  return (
    <>
      <StyledNavbar>
        <StyledBrand>
          <Link to='/'>Brand</Link>
        </StyledBrand>
        <StyledNavLinks>
          <Link to='/addDefect'>Add Defect</Link>
          <Link to='/Second'>Second</Link>
          <Link to='/Third'>Third</Link>
          {isLoggedIn === false ? (
            <Link to='/login'>Login</Link>
          ) : (
            <Link to='/login' onClick={logout}>
              Logout
            </Link>
          )}
        </StyledNavLinks>
      </StyledNavbar>
    </>
  );
};
const mapStateToProps = state => ({
  isLoggedIn: state.authReducer.isLoggedIn,
});
const mapDispatchtoProps = dispatch => {
  return { logout: () => dispatch(logout()) };
};
export default connect(mapStateToProps, mapDispatchtoProps)(Header);
