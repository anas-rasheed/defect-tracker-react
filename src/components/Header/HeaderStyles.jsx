import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export const StyledNavbar = styled.nav`
  height: 3.75rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: var(--oxford-blue);
  /* box-shadow: 0px 6px 0.625rem rgba(157, 160, 252, 1); */
`;

export const StyledBrand = styled.div`
  display: flex;
  align-items: center;
  > a {
    color: var(--cool-gray);
    text-decoration: none;
    padding: 0.625rem;
  }
`;

export const StyledNavLinks = styled.div`
  display: flex;
  height: inherit;
  align-items: center;
  @media screen and (max-width: 786px) {
    display: none;
  }
`;

export const StyledNavLink = styled(NavLink)`
  color: var(--cool-gray);
  position: relative;
  display: flex;
  align-items: center;
  text-align: center;
  height: inherit;
  text-decoration: none;
  padding: 0 1rem;
  transition: all 350ms ease-in-out;
  &::after {
    opacity: 0;
    content: '';
    height: 0.313rem;
    width: 100%;
    display: block;
    background: var(--primary-background);
    position: absolute;
    bottom: 0;
    left: 0;
    transform: translateX(-100px);
  }
  &:hover {
    background: var(--accent);
    opacity: 0.8;
    &::after {
      opacity: 0.8;
      transform: translateX(0);
    }
  }
  &.active {
    &::after {
      opacity: 1;
      transform: translateX(0);
    }
  }
`;
export const StyledMenuContainer = styled.div`
  cursor: pointer;
  display: inherit;
  @media screen and (min-width: 786px) {
    display: none;
  }
`;
export const StyledMenu = styled.div`
  height: 0.125rem;
  background: var(--primary-background);
  position: relative;
  width: 2rem;
  margin: 1rem;
  ::before,
  ::after {
    content: '';
    position: absolute;
    height: 0.125rem;
    width: 2rem;
    background: var(--primary-background);
  }
  ::before {
    transform: translateY(-0.5rem);
  }
  ::after {
    transform: translateY(0.5rem);
  }
`;

export const StyledMenuClose = styled.div`
  height: 0.125rem;
  position: relative;
  width: 1.5rem;
  margin: 1rem;
  cursor: pointer;
  &::before,
  &::after {
    content: '';
    position: absolute;
    height: 0.125rem;
    width: 1rem;
    background: var(--primary-background);
  }
  &::before {
    transform: rotateZ(45deg);
  }
  &::after {
    transform: rotateZ(-45deg);
  }
`;
