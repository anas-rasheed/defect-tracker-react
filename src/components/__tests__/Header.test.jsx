import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../Header/Header';

describe('Header Component', () => {
  it('should render Header component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
