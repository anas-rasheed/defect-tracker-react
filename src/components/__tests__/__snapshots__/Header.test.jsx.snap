// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header component 1`] = `
<Fragment>
  <styled.nav>
    <styled.span>
      <Link
        to="/postingRouter"
      >
        Brand
      </Link>
    </styled.span>
    <styled.span>
      <Link
        to="/"
      >
        First
      </Link>
      <Link
        to="/"
      >
        Second
      </Link>
      <Link
        to="/"
      >
        Third
      </Link>
      <Link
        to="/login"
      >
        Logout
      </Link>
    </styled.span>
  </styled.nav>
</Fragment>
`;
